### PR TITLE
Secure Jaeger UI

### PR DESCRIPTION
**Pull request by radufa**
**Description**

Changes proposed in this pull request:

- Add a "keycloack proxy" as a sidecar to Jaeger 
- Expose Jaeger UI only through its "keycloack" sidecar
- Extend "dex" to support a new static client for Jaeger UI integration 

**Related issue(s)**

Implements #356

### CODE REVIEW

File: resources/core/charts/jaeger/templates/virtualservice.yaml [hunk 1]
```diff
@@ -8,14 +7,26 @@ spec:
   hosts:
   - jaeger.{{ .Values.global.domainName }}
   gateways:
-  - {{ .Values.global.istio.gateway.name }}
+  - core-{{ .Chart.Name }}-gateway
```
Comment (pbochynski) [inline resources/core/charts/jaeger/templates/virtualservice.yaml]: Why can't you use gateway defined for kyma? https://github.com/kyma-project/kyma/blob/master/resources/core/charts/gateway/templates/gateway.yaml

File: resources/core/charts/jaeger/templates/virtualservice.yaml [hunk 2]
```diff
@@ -8,14 +7,26 @@ spec:
...
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: core-{{ .Chart.Name }}-gateway
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
```
Comment (pbochynski) [inline resources/core/charts/jaeger/templates/virtualservice.yaml]: Why do you need http gateway?

File: resources/core/charts/jaeger/templates/virtualservice.yaml (hunk 1 above)
Comment (radufa) [reply resources/core/charts/jaeger/templates/virtualservice.yaml]: We have defined a new “security context” only for Jaeger UI which is independent of the others already existing. Therefore, we have a new “static client” for dex, see the related PR.

Jaeger UI will be used by developers in addition to admins or support. Compromising it should not imply that the others are compromised too.

File: resources/core/charts/jaeger/templates/virtualservice.yaml (hunk 2 above)
Comment (radufa) [reply resources/core/charts/jaeger/templates/virtualservice.yaml]: To be more restrictive in the reachability from outside. All the other ports exposed by Jaeger are available only from calls coming from services running inside of Kyma. We will define Istio authorization rules for them later (we have an issue for this) .

Comment (radufa) [reply resources/core/charts/jaeger/templates/virtualservice.yaml]: Changed

**Comment by abbi-gaurav** [comment]
See also #356

**Pull request merged**